internal/envoy/listener.go

```diff
@@ -220,11 +220,34 @@ func (b *httpConnectionManagerBuilder) AddFilter(f *http.HttpFilter) *httpConnec
 	return b
 }
 
+// Validate runs builtin validation rules against the current builder state.
+func (b *httpConnectionManagerBuilder) Validate() error {
+	filterNames := map[string]struct{}{}
+
+	for _, f := range b.filters {
+		filterNames[f.Name] = struct{}{}
+	}
+
+	// If there's no router filter, requests won't be forwarded.
+	if _, ok := filterNames[wellknown.Router]; !ok {
```

Comment:
I understand your thought here to add this, but seems like a better place for unit tests to cover. There are probably other items that we should be checking on as well.

Reply:
> I understand your thought here to add this, but seems like a better place for unit tests to cover. 

This is by way of defense in depth. I think it's hard for unit tests to cover that every call site does the right thing.

> There are probably other items that we should be checking on as well.

Yep, we can add those here in the future. If you know of things we should check, I can add some extras now.